src/Http/Middleware/HandleErrors.php

```diff
@@ -56,6 +62,24 @@ public function process(Request $request, Handler $handler): Response
         try {
             return $handler->handle($request);
         } catch (Throwable $e) {
+            // If an array of allowlisted exception classes has been provided
+            // (such as when we only want to handle frontend errors), we check
+            // that the error inherits one of these classes. If not, we throw it
+            // to let other handlers up in the middleware pipe handle it.
+            if (is_array($this->errorClasses)) {
+                $handled = false;
+
+                foreach ($this->errorClasses as $errorClass) {
+                    if ($e instanceof $errorClass) {
+                        $handled = true;
+                        break;
+                    }
+                }
+
+                if (! $handled) {
+                    throw $e;
+                }
+            }
```

Comment:
This seems to re-implement logic similar to what the registry already does, and the meaning of the additional parameter doesn't seem very intuitive.